src/main/resources/ddl/achilles/achilles_result_concept_count.sql

```diff
@@ -83,36 +72,56 @@ WHERE analysis_id IN (405, 605, 705, 805, 807, 1805, 1807, 2105)
-FROM @results_schema.achilles_results
-WHERE analysis_id IN (200, 400, 600, 700, 800, 900, 1000, 1300, 1800, 2100, 2200)
+  GROUP BY stratum_2
+), counts_person AS (
+  SELECT stratum_1 as concept_id, MAX (count_value) agg_count_value
+  FROM @results_schema.achilles_results
+  WHERE analysis_id IN (200, 240, 400, 440, 540, 600, 640, 700, 740, 800, 840, 900, 1000, 1300, 1340, 1800, 1840, 2100, 2140, 2200)
     /* analyses:
         Number of persons with at least one visit occurrence, by visit_concept_id
+        Number of persons with at least one visit occurrence, by visit_source_concept_id
         Number of persons with at least one condition occurrence, by condition_concept_id
+        Number of persons with at least one condition occurrence, by condition_source_concept_id
+        Number of persons with death, by cause_source_concept_id
         Number of persons with at least one procedure occurrence, by procedure_concept_id
+        Number of persons with at least one procedure occurrence, by procedure_source_concept_id
         Number of persons with at least one drug exposure, by drug_concept_id
+        Number of persons with at least one drug exposure, by drug_source_concept_id
         Number of persons with at least one observation occurrence, by observation_concept_id
+        Number of persons with at least one observation occurrence, by observation_source_concept_id
         Number of persons with at least one drug era, by drug_concept_id
         Number of persons with at least one condition era, by condition_concept_id
         Number of persons with at least one visit detail, by visit_detail_concept_id
+        Number of persons with at least one visit detail, by visit_detail_source_concept_id
         Number of persons with at least one measurement occurrence, by measurement_concept_id
+        Number of persons with at least one measurement occurrence, by measurement_source_concept_id
         Number of persons with at least one device exposure, by device_concept_id
+        Number of persons with at least one device exposure, by device_source_concept_id
         Number of persons with at least one note by  note_type_concept_id
     */
-GROUP BY stratum_1)
-
+  GROUP BY stratum_1
+), concepts AS (
+  select concept_id as ancestor_id, coalesce(cast(ca.descendant_concept_id as varchar(50)), concept_id) as descendant_id
+  from (
+    select concept_id from counts
+    UNION
+    -- include any ancestor concept that has a descendant in counts
+    select distinct cast(ancestor_concept_id as varchar(50)) concept_id
+    from counts c
+    join @vocab_schema.concept_ancestor ca on cast(ca.descendant_concept_id as varchar(50)) = c.concept_id
+  ) c
+  left join @vocab_schema.concept_ancestor ca on c.concept_id = cast(ca.ancestor_concept_id as varchar(50))
+)
 INSERT INTO @results_schema.achilles_result_concept_count (concept_id, record_count, descendant_record_count, person_count, descendant_person_count)
 SELECT
```

Comment:
I just compared these changes with the latest version of the webapi and here they select DISTINCT, does it matter in our case?

Reply:
Hi Rigbe, thank you for checking. The DISTINCT does not make a difference but still I will add "DISTINCT" and push the change.